tests/integration_tests/data_steward/retraction/retract_deactivated_pids_test.py

```diff
@@ -1,66 +1,114 @@
 
 # Project Imports
 import app_identity
-import bq_utils
 from utils import bq
-import gcs_utils
-from tests import test_util
-from retraction import retract_deactivated_pids
-from constants.cdr_cleaner import clean_cdr as clean_consts
-from sandbox import get_sandbox_dataset_id
-
-TABLE_ROWS_QUERY = 'SELECT * FROM {dataset_id}.__TABLES__ '
-
-EXPECTED_DROPPED_ROWS_QUERY = ('SELECT * '
-                               'FROM {dataset_id}.{table_id} '
-                               'WHERE person_id = {pid} '
-                               'AND {date_column} >= (SELECT deactivated_date '
-                               'FROM {dataset_id}.{pid_table_id} '
-                               'WHERE person_id = {pid})')
-EXPECTED_KEPT_ROWS_QUERY = ('SELECT * '
-                            'FROM {dataset_id}.{table_id} '
-                            'WHERE person_id != {pid} '
-                            'OR(  person_id = {pid} '
-                            'AND {date_column} < (SELECT deactivated_date '
-                            'FROM {dataset_id}.{pid_table_id} '
-                            'WHERE person_id = {pid}))')
-EXPECTED_DROPPED_ROWS_QUERY_END_DATE = (
-    'SELECT * '
-    'FROM {dataset_id}.{table_id} '
-    'WHERE person_id = {pid} '
-    'AND (CASE WHEN {end_date_column} IS NOT NULL THEN {end_date_column} >= '
-    '(SELECT deactivated_date '
-    'FROM `{dataset_id}.{pid_table_id}` '
-    'WHERE person_id = {pid} ) ELSE CASE WHEN {end_date_column} IS NULL THEN '
-    '{start_date_column} >= ( '
-    'SELECT deactivated_date '
-    'FROM `{dataset_id}.{pid_table_id}` '
-    'WHERE person_id = {pid}) END END)')
-EXPECTED_KEPT_ROWS_QUERY_END_DATE = (
-    'SELECT * '
-    'FROM {dataset_id}.{table_id} '
-    'WHERE person_id != {pid} '
-    'OR (person_id = {pid} '
-    'AND (CASE WHEN {end_date_column} IS NOT NULL THEN {end_date_column} < '
-    '(SELECT deactivated_date '
-    'FROM `{dataset_id}.{pid_table_id}` '
-    'WHERE person_id = {pid} ) ELSE CASE WHEN {end_date_column} IS NULL THEN '
-    '{start_date_column} < ( '
-    'SELECT deactivated_date '
-    'FROM `{dataset_id}.{pid_table_id}` '
-    'WHERE person_id = {pid}) END END))')
-INSERT_PID_TABLE = ('INSERT INTO {dataset_id}.{pid_table_id} '
-                    '(person_id, deactivated_date) '
-                    'VALUES{person_research_ids}')
+from retraction import retract_deactivated_pids as rdp
+import sandbox as sb
+from common import JINJA_ENV
+
+DEACTIVATED_PIDS_SCHEMA = [{
+    "type": "integer",
+    "name": "person_id",
+    "mode": "required",
+    "description": ""
+}, {
+    "type": "date",
+    "name": "deactivated_date",
+    "mode": "nullable",
+    "description": ""
+}]
+
+DEACTIVATED_PIDS = JINJA_ENV.from_string("""
+INSERT INTO `{{deact_table.project}}.{{deact_table.dataset_id}}.{{deact_table.table_id}}` 
+(person_id, deactivated_date) 
+VALUES
+(1,'2009-07-25'),
+(2,'2009-03-14'),
+(3,'2009-11-18'),
+(4,'2009-11-25'),
+(5,'2009-09-20')
+""")
```

Comment:
I made this change and used `NO_CONTACT` as suspension status, but it looks like `NOT_SUSPENDED` is also possible. What should the behavior be for such participants?

Reply:
I think `NOT_SUSPENDED` = "Currently active participant"
and
`NO_CONTACT` = "the participant deactivated on the provided date".  So, the `_deactivated_participants` table should be storing data for with a status of `NO_CONTACT`, because that should be all the is requested via the API.

Reply:
I see, so there won't be any participants with `NOT_SUSPENDED` in the `_deactivated_participants`. Thanks, that clears it up.